app/javascript/controllers/better_together/message_visibility_controller.js

```diff
@@ -0,0 +1,56 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="message-visibility"
+export default class extends Controller {
+	connect() {
```

Comment:
You may want to add a disconnect() method to handle clean up of the observer (or at least add a TODO) if for some reason the message is removed from the DOM before the observer is detached. It's pretty cool to see this in action, though!!